dask/tests/test_config.py

```diff
@@ -61,6 +60,8 @@ def test_merge():
 
 
 def test_collect_yaml_paths():
+    yaml = pytest.importorskip("yaml")
```

Comment:
As in, disable the whole module if yaml isn't installed?

Reply:
My understandiing is that we don't actually test the stdlib only case in our CI.  I would default to just leaving this as it was before.  Presumably if people don't have yaml installed and they try to run tests then they would get an `ImportError`. Personally I'm ok with this.  I don't think that this policy has caused significant pain so far, so I personally wouldn't spend much effort changing it.

Reply:
Maybe I should ask though, did this policy cause you significant pain for some reason?  I'm not sure I fully understand your motivation here.

Reply:
For yaml, no. For cloudpickle, yes, and it's what caused me to write this PR to begin with: https://github.com/pydata/xarray/issues/3401

Reply:
I am now disabling test_config.py if yaml isn't installed. 

I agree that nobody cares if tests accidentally fall over locally because of an undeclared dependency. Users that don't need the dask config files (most beginners, in fact) must however be guaranteed that everything still works if PyYAML is not installed. This is covered by the two new CI test suites.
Again, the alternative to all of this would just to set PyYAML as a required dependency.